src/js/__tests__/app.test.js

```diff
@@ -101,8 +106,103 @@ describe('사용자가 유효하지 않은 값을 입력했을 경우, 에러를
   test(`사용자가 ${autoComma(
     LOTTO.PRICE_PER_TICKET,
   )}원 단위로 금액을 투입하지 않았을 경우 에러를 발생시킨다.`, () => {
-    const invalidMoney = 1500;
+    const invalidMoney = LOTTO.PRICE_PER_TICKET + LOTTO.PRICE_PER_TICKET / 2;
 
     validateErrorMessage(invalidMoney, EXCEPTION.INVALID_UNIT);
   });
 });
+
+describe('당첨 번호를 입력하면, 로또에 대한 통계를 확인할 수 있다.', () => {
+  const lottoPrize = new LottoPrize();
+
+  beforeEach(() => {
+    lottoPrize.initialize();
+  });
+
+  const calculateLottoPrizeCount = (
+    lottoPrizeNumbers,
+    lottoBonusNumber,
+    userLottoNumbers,
+  ) => {
+    userLottoNumbers.forEach((numbers) => {
+      lottoPrize.countPrize(
+        returnSameNumberCount(numbers, lottoPrizeNumbers),
+        numbers,
+        lottoBonusNumber,
+      );
+    });
+  };
+
+  test('몇 개의 로또가 당첨되었는지 확인할 수 있다.', () => {
+    const lottoPrizeNumbers = [1, 2, 3, 4, 5, 6];
+    const lottoBonusNumber = 7;
+    const userLottoNumbers = [
+      [1, 2, 3, 4, 5, 6], // first prize
+      [1, 2, 3, 4, 5, 7], // second prize
+      [1, 2, 3, 4, 5, 8], // third prize
+      [1, 2, 3, 4, 8, 9], // fourth prize
+      [1, 2, 3, 8, 9, 10], // fifth prize
+    ];
+
+    calculateLottoPrizeCount(
+      lottoPrizeNumbers,
+      lottoBonusNumber,
+      userLottoNumbers,
+    );
+
+    PRIZE_NAMES.forEach((prizeName) => {
+      expect(lottoPrize.prizeCount[prizeName]).toBe(1);
+    });
+  });
+
+  test('로또 당첨금액에 대한 수익률을 계산할 수 있다.', () => {
+    const inputMoney = LOTTO.PRICE_PER_TICKET * 2;
+    const lottoPrizeNumbers = [1, 2, 3, 4, 5, 6];
+    const lottoBonusNumber = 7;
+    const userLottoNumbers = [
+      [1, 2, 3, 4, 8, 9], // fourth prize
+      [1, 2, 3, 8, 9, 10], // fifth prize
+    ];
+    const prizeMoney = 55000;
+    const correctRateOfReturn = ((prizeMoney - inputMoney) / inputMoney) * 100;
+
+    calculateLottoPrizeCount(
+      lottoPrizeNumbers,
+      lottoBonusNumber,
+      userLottoNumbers,
+    );
+    lottoPrize.calculateRateOfReturn(inputMoney);
+
+    expect(lottoPrize.rateOfReturn).toBe(correctRateOfReturn);
+  });
+});
+
+describe('당첨 번호를 잘못 입력하면 오류를 발생시킨다.', () => {
+  const validateErrorMessage = (invalidNumbers, errorMessage) => {
+    try {
+      validatePrizeNumber(invalidNumbers);
+    } catch (error) {
+      expect(error.message).toBe(errorMessage);
+    }
+  };
+
+  test('사용자가 당첨 번호나 보너스 번호를 전부 입력하지 않은 경우 오류를 발생시킨다.', () => {
+    const invalidPrizeNumbers = [1, 2, 3, 4, 5, NaN];
+    const invalidBonusNumber = NaN;
+
+    validateErrorMessage(
+      [...invalidPrizeNumbers, invalidBonusNumber],
+      EXCEPTION.BLANK_PRIZE_NUMBER,
+    );
```

Comment:
```suggestion
    expect(() =>
      validatePrizeNumber([...invalidPrizeNumbers, invalidBonusNumber]),
    ).toThrowError(EXCEPTION.BLANK_PRIZE_NUMBER);
```

요렇게도 에러메세지를 통해 테스트할 수 있어요 👀

Reply:
앗! `toThrowError`를 사용하면 `try-catch`로 에러를 받고 에러메시지를 직접 비교할 필요가 없겠네요!!
많이 알아갑니다... 🥲 정말 검색을 생활화해야겠어요 흑흑